### PR TITLE
Announce deprecation of `.flutter-plugins`.

### PR DESCRIPTION
**Pull request by matanlurey**
Work towards https://github.com/flutter/flutter/issues/48918.

### CODE REVIEW

**Comment by matanlurey** [comment]
Done.

File: src/content/release/breaking-changes/flutter-plugins-configuration.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  }
+}
+```
+
+Having both files output is a source of technical debt that complicates adding
+and supporting the `flutter` CLI.
+
+## Migration guide
+
+Most end users of Flutter do not parse or use this file, but build artifacts
```
Comment (sfshaza2) [inline src/content/release/breaking-changes/flutter-plugins-configuration.md:61]:
```suggestion
Most Flutter developers don't parse or use this file, but build artifacts
```

File: src/content/release/breaking-changes/flutter-plugins-configuration.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+such as `settings.gradle` generated by older invocations of
+`flutter create --platforms android` may still reference `.flutter-plugins` and
+must be updated to a newer build script.
```
Comment (sfshaza2) [inline src/content/release/breaking-changes/flutter-plugins-configuration.md:64]:
```suggestion
(such as `settings.gradle`), as generated by older invocations of
`flutter create --platforms android`. These legacy files might still reference `.flutter-plugins` and
must be updated to a newer build script.
```

File: src/content/release/breaking-changes/flutter-plugins-configuration.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+file, and only output `.flutter-plugins-dependencies`. Tools and build scripts,
+such as Gradle configurations (for Android apps) that rely on the presence of
+`.flutter-plugins` will need to be updated.
+
+## Background
+
+[In 2019][PR 45379] `.flutter-plugins-dependencies` was added as a newer file
+format that replaces `.flutter-plugins`.
+
+So a file that looks something like this:
```
Comment (sfshaza2) [inline src/content/release/breaking-changes/flutter-plugins-configuration.md:21]:
```suggestion
So a file that looked something like this:
```

**Review by sfshaza2 — APPROVED** [review]
A bit of smoothing out the English. Once addressed, lgtm.

**Review by johnmccutchan — APPROVED** [review]

File: src/content/release/breaking-changes/flutter-plugins-configuration.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    }
+  },
+  "flutter": {
+    "frameworkRevision": "3a0f99d4f2",
+    "channel": "stable"
+  }
+}
+```
+
+Having both files output is a source of technical debt that complicates adding
```
Comment (reidbaker) [inline src/content/release/breaking-changes/flutter-plugins-configuration.md:56]: adding what? adding to?

**Review by reidbaker — APPROVED** [review]

File: src/content/release/breaking-changes/flutter-plugins-configuration.md (hunk 4 above)
Comment (matanlurey) [reply src/content/release/breaking-changes/flutter-plugins-configuration.md]: Oops, done!

**Review by reidbaker — APPROVED** [review]

**Pull request merged**